### ISSUE TITLE
In vi-copy-mode keep highlighted text when scrolling

### ISSUE THREAD

**Issue by ksrb** [tmux/tmux#950]
It would be nice if in copy-mode the text highlighting while searching could be retained.

**Steps to reproduce**
```bash
man tmux # or anything really
```
<kbd>prefix + /</kbd> - begin search
<kbd>tmux</kbd> - search for 'tmux'
<kbd>j</kbd> - move cursor down and highlighting stops 😞

**System information**
```bash
uname -sp && tmux -V && echo $TERM
CYGWIN_NT-10.0 unknown
tmux 2.4
screen-256color
```

**Comment by nicm** [comment]
No, when you move the cursor or issue another command, search is cancelled.

**Comment by ksrb** [comment]
Hmm could I make this a feature request then 😄? 

Could change the title to: Option to retain/remove the search highlight.

**Comment by nicm** [comment]
No this is how emacs works and I don't see a need for multiple behaviours
here.

**Comment by ksrb** [comment]
Tried it in emacs yup that's how it works, though there are plugins in emacs to make highlighting [persistent](https://github.com/juanjux/evil-search-highlight-persist) 😉.

Just wondering how feature request are taken here. Some of the [open source projects](https://github.com/VSCodeVim/Vim/blob/master/.github/ISSUE_TEMPLATE.md) I've seen are using a simple voting systems by 👍/👎 a issue, which the contributors use to prioritize features. Any chance on adopting something similar or should I just close the issue?

**Comment by nicm** [comment]
If someone writes this so it works only with vi keys, I'll look at it - but I'm not going to do it, so I think there is no need to keep this issue open. Thanks

**Comment by ericpruitt** [comment]
@nicm:

> If someone writes this so it works only with vi keys, I'll look at it

I created a patch that implements this feature when copy mode is using Vi keybindings. Could you please take a look at it and let me know if there are any obvious issues with my implementation?

**Comment by nicm** [comment]
Could you open a new issue or a PR please, I will probably not be able to
look at it until third week of December and if it is a new issue I am more
likely to remember.

**Comment by ericpruitt** [comment]
In that case, I'll just create a pull request next month. I've been using this patch for literally less than an hour. Since I run tmux by default within my graphical emulator, the code should be well exercised by then.

**Comment by cangermueller** [comment]
I have still the same problem: search highlighting disappears when scrolling in vi-copy-mode.

> uname -sp && tmux -V && echo $TERM
Linux unknown
tmux 2.6
xterm-256color

@ericpruitt, is your patch integrated into tmux-2.6?

**Comment by ericpruitt** [comment]
> @ericpruitt, is your patch integrated into tmux-2.6?

No. Look at #1194.

**Comment by ericpruitt** [comment]
This is fixed as of https://github.com/tmux/tmux/commit/c2fde58701d3d0a55ba15ba6004df7ca009175b9, https://github.com/tmux/tmux/commit/87a11a921455538ba4594e0852439f84e7b84a4c and https://github.com/tmux/tmux/commit/58f870ef6ec4b9dfb5646cf4eafabaac6ea125bc

**Comment by tmpm697** [comment]
@ericpruitt is there a tmux's option to enable this? I find it feature extremely useful when working with tmux.

**Comment by ericpruitt** [comment]
No, it's only enabled with Vi copy mode bindings because of https://github.com/tmux/tmux/issues/950#issuecomment-305020541.

**Comment by tmpm697** [comment]
@ericpruitt will it enabled by default when I set `mode-keys` to vi in `tmux.conf`?

I had `mod-keys` to `vi` but highlight search still disappear.

**Comment by ericpruitt** [comment]
The change was not included in the 3.0 release. You'll have to recompile from master or backport the patch.

**Comment by tmpm697** [comment]
@ericpruitt Thanks, hope it will be released soon on archlinux.
